docs/content/policy-language.md

```diff
@@ -975,6 +975,65 @@ s(5, 3)
 ```live:eg/double_function_define_undefined/2:output:expect_undefined
 ```
 
+#### Function overloading
+Rego does not currently support the overloading of functions by the number of parameters. If two function definitions are given with the same function name but different numbers of parameters, a compile-time type error is generated.
+
+```live:eg/function_overloading_error:module
+r(x) := result if {
+    result := 2*x
+}
+
+r(x, y) := result if {
+    result := 2*x + 3*y
+}
```

Comment:
I might be the odd one here, but I think it might be slightly more readable if condensed:

```rego
r(x) := 2*x

r(x, y) := 2*x + 3*y
```
WDYT?

Reply:
Happy to use the more condensed way. I was doing it this way to stay consistent with the pre-existing function definition examples in the same section eg line 920.

Should I change those too?

Reply:
> Should I change those too?

Hmm I'll go with what you prefer, I'm not sure what the better exposition is here. The "condensed" form could be confusing; and what we have here is easier to use as a starting point for your own functions... few of them (I presume) would have a condensed representation.

I suppose it's just that `f(x) := result if { result := ... }` makes _me_ think that it was something like a relation; when really, `f(x)` just is that result, since that body cant fail... 🤔